skunkworks_crow/src/main/java/org/odk/share/preferences/SettingsPreference.java

```diff
@@ -159,4 +155,25 @@ private void showPasswordDialog() {
         AlertDialog alertDialog = builder.create();
         alertDialog.show();
     }
+
+    public void chooseDirectory() {
+        Intent i = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE);
+        i.addCategory(Intent.CATEGORY_DEFAULT);
+        startActivityForResult(Intent.createChooser(i, getString(R.string.choose_directory)), DIRECTORY_REQUEST_CODE);
+    }
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        switch (requestCode) {
+            case DIRECTORY_REQUEST_CODE:
+                try {
+                    String filePath = (data.getData()).getPath();
+                    filePath = filePath + getString(R.string.directory_odk);
```

Comment:
`filePath = filePath + getString(R.string.directory_odk);` can be written as  `filePath = filePath + getString(R.string.directory_odk);`